### ISSUE TITLE
WebGL2: Differences from OpenGL ES 3.

### ISSUE THREAD

**Issue by grorg** [KhronosGroup/WebGL#543]
Fill in the section listing differences between WebGL 2.0 and OpenGL ES 3.0

**Comment by Kuranes** [comment]
Not sure where to post, but seems a difference not listed or clarification not made, but WebGL2 accept reversed depthrange (as OpenGL ES 3.0 does see https://www.khronos.org/opengles/sdk/docs/man3/html/glDepthRangef.xhtml ) whereas WebGL1  didn't ( https://www.khronos.org/registry/webgl/specs/1.0/#VIEWPORT_DEPTH_RANGE ) ?

**Comment by zhenyao** [comment]
I think WebGL2 inherits all WebGL1 limits unless it's explicitly lifted in the spec, so in WebGL2 revered depth range is also forbidden.

**Comment by kenrussell** [comment]
@Kuranes @zhenyao We tried to get some clarification on whether Direct3D 11 removed a limitation in this area but I'm not sure we got a firm answer. Please consider that the restriction is still there and file a new issue about it with a test case demonstrating what should happen if it's working correctly. Thanks. CC @null77 @vonture

**Comment by vonture** [comment]
I can confirm that D3D11 does not allow inverted depth range.  We have an idea that may allow us emulate this in the future but we haven't prototyped it ([anglebug.com/1605](anglebug.com/1605)).

**Comment by Kuranes** [comment]
Strange, D3D11 does allow inversed depth range: see the test app PrTEST.zip from https://mynameismjp.wordpress.com/2010/03/22/attack-of-the-depth-buffer/ blogpost talking about that exact topic ( Samples images & dx11 source code )

@kenrussell thanks, added a separate issue about just that topic https://github.com/KhronosGroup/WebGL/issues/2197

**Comment by vonture** [comment]
This bug may have been filed against D3D9, I had assumed it was D3D11.  Let me verify locally.

**Comment by vonture** [comment]
Just verified, D3D11 does NOT allow inverted depth range by simply setting inverting the near and far planes (as the bug said).  The samples provided all implement the functionality by using a regular 0..1 depth range but use a projection matrix that inverts the Z and inverting the depth test.

**Comment by Kuranes** [comment]
Indeed. sorry, I mixed up things. the "reveresed trick" on desktop OpenGL is done this with the widely-supported ARB_clip_control extension (now also core in OpenGL 4.5 as glClipControl). Unfortunately, in GL ES you're out of luck.
